### PR TITLE
Fail the build when a k8s deployment was requested by API server can't be reached

### LINKED ISSUE #9129: when cluster is not available should build not fail when deploy requested ?
**Describe the bug**
have cluster configured in kube but the cluster being down/not reachable:
run `mvn clean package -Dquarkus.kubernetes.deploy=true`

and you get output like this:

```shell
[WARNING] Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
[WARNING] Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
[ERROR] [io.quarkus.kubernetes.deployment.KubernetesDeploy] Although a Kubernetes deployment was requested, it will however not take place because there was an error during communication with the API Server: An error has occurred.
[WARNING] Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
[WARNING] Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
[INFO] [org.jboss.threads] JBoss Threads version 3.1.1.Final
[WARNING] Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
[WARNING] Error reading service account token from: [/var/run/secrets/kubernetes.io/serviceaccount/token]. Ignoring.
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building thin jar: /Users/max/Dropbox/code/openshift-deployment/target/simple-1.0.0-SNAPSHOT-runner.jar
[INFO] Initializing dekorate session.
[INFO] Default s2i build generator....
[INFO] Generating manifests.
[INFO] Processing kubernetes configuration.
[INFO] Processing openshift configuration.
[INFO] Processing s2i configuration.
[INFO] Closing dekorate session.
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 8761ms
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

**Expected behavior**
I would have expected a BUILD FAILURE here.

**Actual behavior**
build marked as success despite there are errors.

### PR DESCRIPTION
**Pull request by geoand**
…t be reached

Fixes: #9129

### CODE REVIEW

**Comment by geoand** [comment]
This is technically a breaking change...

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java [hunk 1]
```diff
@@ -32,25 +31,28 @@ public boolean getAsBoolean() {
...
-                    log.warn(message);
-                } else {
-                    log.error(
-                            "Although a Kubernetes deployment was requested, it will however not take place because there was an error during communication with the API Server: "
-                                    + e.getMessage());
-                }
-                alreadyWarned = true;
+            if (e.getCause() instanceof SSLHandshakeException) {
+                throw new RuntimeException(
+                        "Although a Kubernetes deployment was requested, it however cannot take place because the API Server certificates are not trusted. The certificates can be configured using the relevant configuration propertiers under the 'quarkus.kubernetes-client' config root, or \"quarkus.kubernetes-client.trust-certs=true\" can be set to explicitly trust the certificates (not recommended)",
```
Comment (maxandersen) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java:48]: should this not also include the masterURL?

**Comment by maxandersen** [comment]
> This is technically a breaking change...

was it ever expected that the build pass ?

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java (hunk 1 above)
Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesDeploy.java:48]: Ah yeah, I missed it

**Comment by geoand** [comment]
> > This is technically a breaking change...
> 
> was it ever expected that the build pass ?

It shouldn;t have passed, it was a mistake to let it pass when the deployment didn't work

**Comment by geoand** [comment]
PR updated. @iocanel can you also take a look since I had to make a few more changes than I thought were necessary.

**Review by iocanel — APPROVED** [review]
Lgtm

**Comment by geoand** [comment]
@maxandersen OK with you?

**Review by maxandersen — APPROVED** [review]
lgtm

**Pull request merged**